devito/ir/clusters/algorithms.py

```diff
@@ -176,7 +177,8 @@ def guard(clusters):
                 if cd.condition is None:
                     condition.append(CondEq(cd.parent % cd.factor, 0))
                 else:
-                    condition.append(cd.condition)
+                    cond = lower_exprs(cd.condition).pop()
```

Comment:
I don't see how can this work?  `cd.condition` is an expression (not a list of expressions), and you pass it to `lower_exprs` which, in turn, performs `for expr in expressions`, but `expressions` as I just wrote is not a list

in `lower_exprs`, you probably should:
```
for expr in expressions: -> for expr in as_tuple(expressions):
```
and, at the end, `return` as follows
```
if isinstance(expressions, Iterable):
    return processed  # a list
else:
    assert len(processed) == 1
    return processed.pop()
```

IOW, if `lower_exprs` is given a list, it should return a list; if it's given a single expression, it should return a single expression, not a list. This makes `lower_exprs` more flexible while the caller won't need this horrible `.pop()`

Reply:
agree

Reply:
Fixed